stereo/plots/paga_plot.py

```diff
@@ -0,0 +1,449 @@
+                    ).coords
+                except AttributeError:  # hack for empty graphs...
+                    pos_list = g.layout(layout, seed=init_coords, **layout_kwds).coords
+            pos = {n: [p[0], -p[1]] for n, p in enumerate(pos_list)}
+        if len(pos) == 1:
+            pos[0] = (0.5, 0.5)
+        pos_array = np.array([pos[n] for count, n in enumerate(nx_g_solid)])
+        #stereo_exp_data.tl.result['paga']['pos'] = pos_array
+        return pos_array
+    
+    
+    def paga_plot(
+        self, 
+        sdata,
+        adjacency: str = 'connectivities',
+        threshold: Optional[float] = None,
+        layout: Optional[_Layout] = None,
+        random_state: int = 0,
+        #labels: Union[str, Sequence[str], Mapping[str, str], None] = None,
+        cmap: Optional[str]='tab20',
+        ax: Optional[plt.Axes] = None
+    ):
+        """
+        abstract paga plot for the paga result.
+        param sdata:StereoExpData with paga result
+        param adjacency: keyword to use for paga or paga tree
+        threshold: prune edges lower than threshold
+        param layout: the method to layout each node
+        param random_state: to control the random initialization
+        param cmap: colormap to use, default with tab20
+        param ax: subplot to plot.
+        return: paga abstract graph
+        """
+        # calculate node positions 
+        adjacency_mat = sdata.tl.result['paga'][adjacency].copy()
+        if threshold is None:
+            threshold = 0.01  # default threshold
+        if threshold > 0:
+            adjacency_mat.data[adjacency_mat.data < threshold] = 0
+            adjacency_mat.eliminate_zeros()
+    
+        pos = self._compute_pos(adjacency_mat, layout=layout, random_state=random_state)
+        sdata.tl.result['paga']['pos'] = pos
+    
+            
+        # network
+        G = pd.DataFrame(adjacency_mat.todense())
+        ct_list = sdata.cells.to_df()[sdata.tl.result['paga']['groups']].cat.categories
+        G.index = ct_list
+        G.columns = ct_list
+        Edges = nx.from_pandas_adjacency(G).edges()
+        Nodes2pos = dict(zip(ct_list, list(pos)))
+        
+        # define colors
+        ct_list = sdata.cells[sdata.tl.result['paga']['groups']].cat.categories
+        color_list = plt.get_cmap(cmap, len(ct_list))
+        ct2color = {x:color_list.colors[i] for i,x in enumerate(ct_list)}
+        
+        # plotting
+        if ax == None:
+            _, ax = plt.subplots(1)
+        
+        ax.scatter(pos[:, 0], pos[:, 1], c = color_list.colors, zorder=1)
+        for i in range(len(ct_list)):
+            ax.text(pos[i, 0], pos[i, 1], s=ct_list[i], zorder=2)
+        for i,j in Edges:
+            xi, yi = Nodes2pos[i]
+            xj, yj = Nodes2pos[j]
+            ax.plot([xi,xj], [yi, yj], color='black', zorder=0)
+        ax.set_xticks([])
+        ax.set_yticks([])
+        return ax
```

Comment:
In order to download plot on notebook(this function has been implemented), It should return a object of Matplotlib Figure.